.eslintrc.js

```diff
@@ -32,4 +32,35 @@ module.exports = {
     '@next/next/no-img-element': ['off'],
     'linebreak-style': ['error', os.platform() === 'win32' ? 'windows' : 'unix'],
   },
+  overrides: [
+    {
+      files: ['**/*.ts', '**/*.tsx'],
+      extends: [
+        'next',
+        'airbnb',
+        'airbnb-typescript',
+        'plugin:cypress/recommended',
+        'plugin:@typescript-eslint/eslint-recommended',
+        'plugin:@typescript-eslint/recommended',
+      ],
+      parser: '@typescript-eslint/parser',
+      parserOptions: {
+        ecmaVersion: 12,
+        sourceType: 'module',
+        project: ['./tsconfig.json'],
+      },
+      plugins: ['@typescript-eslint', 'eslint-plugin-jsx-a11y', 'cypress'],
+      rules: {
+        '@typescript-eslint/no-unused-vars': [
+          'warn',
+          {
+            varsIgnorePattern: '^_',
+            argsIgnorePattern: '^_',
+          },
+        ],
+        '@typescript-eslint/no-empty-function': ['warn'],
+        'prefer-const': 'error',
+      },
+    },
+  ],
```

Comment:
this `overrides` section doesn't really need to be here - let's prioritize including all these configurations in the original body of this eslintrc file

Reply:
I added the overrides because the app is yet to go full on typescript, so that section takes care of the linting for the ts files.There'd be errors if only the ts linting is used since most of the app is still common js.

Reply:
you can have both eslint and ts linting running at the same time, we use that for our other open source project - i don't think it's the best to duplicate entries in this file if possible